lib/livebook_web/utils.ex

```diff
@@ -0,0 +1,34 @@
+defmodule LivebookWeb.Utils do
+  @moduledoc "Utility functions for web-focused tasks"
+
+  # Converts human-readable strings to strings which can be used
+  # as HTML element IDs (compatible with HTML4+)
+  #
+  # At the same time duplicate IDs are enumerated to avoid duplicates
+  @spec names_to_ids([binary], MapSet.t()) :: list
+  def names_to_ids(names, used_ids \\ MapSet.new())
```

Comment:
This `user_ids` default argument implies we get public `names_to_ids/1` and `names_to_ids/2` functions. As this is an internal detail it's generally better to do:

```elixir
def names_to_ids(names) do
  do_names_to_ids(names, MapSet.new())
end

# ...
```

Reply:
Yeah, I was trying to avoid making a `do_` function.  I realize it's a very standard convention, but creating functions like that feels arbitrary so I avoid it when I can.  But I also agree that having the "hidden" section argument isn't great either.

Regardless, I changed it to use `Stream.transform` which is so much nicer overall (❤️❤️ to Elixir re: how there's always a general helper that I need, I just sometimes need to find it 😄 )